### PR TITLE
Add XQuery template for Oxygen

### PR DESCRIPTION
**Pull request by AirQuick**
This pull request adds an XQuery template for Oxygen.
With this change and `xspec.framework` enabled, Oyxgen menu - **File** - **New** will look like this:

xq-tmpl

The past directory structure of this repository suggests that the Oxygen template files should go in `editors/oxygen/templates/`, but that doesn't seem to fit very nicely into `[Oxygen installation]/frameworks/xspec`. Also `xspec.framework` already has this field:

```xml
<field name="templatesLocations">
	<String-array>
		<String>${frameworkDir}/templates</String>
	</String-array>
</field>
```

That's why the files are placed in `templates/`.

### CODE REVIEW

File: templates/xquery.xspec [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+    * Update /x:description/@query which should be the namespace URI of the XQuery module being tested.
+    * Update /x:description/@query-at which should point to the XQuery module file being tested.
+    * Write your test scenario. See https://github.com/xspec/xspec/wiki/Writing-Scenarios for details.
+-->
+<x:description xmlns:x="http://www.jenitennison.com/xslt/xspec"
+    query="http://www.example.com/my-module" query-at="uri/to/my-module.xquery">
```
Comment (galtm) [inline templates/xquery.xspec]: Should `uri/to/my-module.xquery` say something like `path/to/my-module.xquery` to reduce confusion with the `query` attribute?

Comment (AirQuick) [reply templates/xquery.xspec]: Good point. Fixed now.

**Comment by AirQuick** [comment]
> One question or possibly the subject of a new pull request: Should the `templates` folder also include the XSLT Unit Test and Schematron Unit Test templates that are currently available in Oxygen under File > New > "New Document"? That way, users can see all three templates in the same place. (If you add those, can you fix the label that has a typo, "speciffic"?)

That will also help users of older versions of Oxygen. For example, 19.x does not have the Schematron template.
For Schematron, I opened #745. Can you take a look at it?
For XSLT, I'm not sure what to do. The built-in XSLT template has a custom dialog which cannot be replaced instantly.

**Comment by galtm** [comment]
> older versions of Oxygen

Good point!

I just took a quick look at #745. I'm not sure what you mean about the custom dialog in the built-in XSLT template. If you're looking for input, can you clarify?

**Comment by AirQuick** [comment]
If you click the **XSLT Unit Test** template in **File** - **New** dialog, the **Customize** button is enabled and it takes you to a custom dialog. Looks like they call it [_New XSLT Unit Test Wizard_](https://www.oxygenxml.com/doc/versions/21.1/ug-editor/topics/xslt-unit-test-xspec.html).

**Comment by galtm** [comment]
Thanks. I never noticed that Customize button! I usually double-click the "XSLT Unit Test" listing on top.

>custom dialog which cannot be replaced instantly
Is there a technical barrier, or is it just shortage of time?

**Comment by AirQuick** [comment]
> Is there a technical barrier, or is it just shortage of time?

Mainly a technical barrier. Customization of that kind requires Java.

**Comment by galtm** [comment]
> Mainly a technical barrier. Customization of that kind requires Java.

OK, I'll think about this a bit more and see if I get any ideas for alternatives.

**Comment by galtm** [comment]
> For XSLT, I'm not sure what to do. The built-in XSLT template has a custom dialog which cannot be replaced instantly.

Here are my ideas, where I lean toward the first one.

- Do nothing for now. Maybe browsing is less common than typing "XSpec" or "Unit Test" into the edit field, in which case the user can easily find any of the 3 templates. Pro: Requires no additional work and does not introduce new problems. Con: Does not address categorization inconsistency.
- Create our own static template and label it as something like "XSLT Unit Test (without wizard)". Pro: XSpec framework has 3 listings. Cons: Might be confusing to have two templates with similar names, especially when you get a different uncustomized template if you click Customize for the built-in template, leave the prompts empty, and click Create. Also, the labeling itself might be confusing. I'm not crazy about any of the labeling ideas I had: without wizard; uncustomized; plain template.

**Comment by AirQuick** [comment]
Thanks for sharing your thoughts.

I'm for this one

> * Do nothing for now.

mostly because of this

> typing "XSpec" or "Unit Test" into the edit field, in which case the user can easily find any of the 3 templates.

**Pull request merged**